pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -223,6 +223,8 @@ type Controller struct {
 	// if set to true, NodeController will taint Nodes based on its condition for 'NetworkUnavailable',
 	// 'MemoryPressure', 'OutOfDisk' and 'DiskPressure'.
 	taintNodeByCondition bool
+
+	nodeUpdateEvent cache.FIFO
```

Comment:
@k82cn - thanks for looking into that.

Unfortunately this PR I think is not correct, because in theory what may happen is that:
- update U1 on node N1 comes
- second update U2 on node N1 comes
- those two are picked at the same time by two workers
- the second one is send first to apiserver
- the first one is sent earlier
- and we end up in a bad end state

Instead you should probably do sth similar that I did in https://github.com/kubernetes/kubernetes/pull/64431

Reply:
oh, yes, you're right :) I'm going to update it tomorrow :).

Reply:
@wojtek-t didnt understand the solution in 64431, can you elaborate how are you solving this issue you mention above ?

Reply:
It's ensuring that all events for a given node are processed by the same worker.
Since the worker is processing events in the correct order, you get the guarantee that no race can happen.

Reply:
found it this line would ensure the hash for the same node is always the same as long as the worker size doesnt change